pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/rules/ItemsRule.scala

```diff
@@ -15,33 +11,34 @@ import cats.data.NonEmptyList
   *
   * * Sierra - Single items
   *   * METS works
-  *   * Miro works
+  *   * Single Miro works, only if the Sierra work has workType Picture/Digital Image/3D Object
   * * Sierra - Multi item
   *   * METS works
   */
 object ItemsRule extends FieldMergeRule with MergerLogging {
+  import WorkPredicates._
+
   type FieldData = List[Item[Unminted]]
 
   override def merge(
     target: UnidentifiedWork,
     sources: Seq[TransformedBaseWork]): FieldMergeResult[FieldData] = {
-    val rules =
-      List(
-        mergeIntoCalmTarget,
-        mergeMetsIntoSingleItemSierraTarget,
-        mergeMiroIntoSingleItemSierraTarget,
-        mergeIntoMultiItemSierraTarget)
-
     val items =
       mergeIntoCalmTarget(target, sources)
         .orElse(mergeMetsIntoSingleItemSierraTarget(target, sources)
-          .orElse(mergeMiroIntoSingleItemSierraTarget(target, sources)))
+          .orElse(mergeSingleMiroIntoSingleItemSierraTarget(target, sources)))
         .orElse(mergeIntoMultiItemSierraTarget(target, sources))
         .getOrElse(target.data.items)
 
-    val mergedSources = sources.filter { source =>
-      rules.exists(_(target, source).isDefined)
-    } ++ findFirstLinkedDigitisedSierraWorkFor(target, sources)
+    val mergedSources = (
+      List(
+        mergeIntoCalmTarget,
+        mergeMetsIntoSingleItemSierraTarget,
+        mergeSingleMiroIntoSingleItemSierraTarget,
+        mergeIntoMultiItemSierraTarget)
+        .flatMap(_.mergedSources(target, sources)) ++
+        findFirstLinkedDigitisedSierraWorkFor(target, sources)
+    ).distinct
```

Comment:
nit: the way this is indented / laid out makes it pretty hard to parse for me (dunno if this is scalafmt being a nuisance or not)